packages/pluggableWidgets/rich-text-web/rollup.config.js

```diff
@@ -0,0 +1,72 @@
+import { join, extname } from "path";
+import json from "@rollup/plugin-json";
+import replace from "rollup-plugin-re";
+import command from "rollup-plugin-command";
+import copy from "recursive-copy";
+import through from "through2";
+
+export default args => {
+    const outDir = join(__dirname, "dist/tmp/widgets/ckeditor");
+    const result = args.configDefaultConfig;
+    return result.map((config, index) => {
+        // Only copying on first step is enough
+        if (index === 0) {
+            config.plugins.push(copyCKEditorDirToDist(outDir));
+        }
+        /* Need to replace icon path from bundled js files, it's step 0 and 1, 
+        tried to find much safer comparison but there are no uniqueness, so the best idea is to replace the path by index */
+        if (index === 0 || index === 1) {
+            config.plugins.push(replaceIconPath());
+        }
+        /* this step is required by sanitize-html library */
+        config.plugins.push(json());
+        return { ...config };
+    });
+};
+
+function copyCKEditorDirToDist(outDir) {
+    return command([
+        async () => {
+            return copy("../../../node_modules/ckeditor4/", outDir, {
+                transform: src => {
+                    /* we need to empty every single css inside the folder to avoid duplicate bundling,
+                     because if the file even not imported anywhere it will be compiled inside RichText.js and RichText.mjs */
+                    if (extname(src) === ".css") {
+                        return through((chunk, enc, done) => {
+                            done(null, "");
+                        });
```

Comment:
We faced file loading issue in the process of integration. **CDN vs local path.**
When we defined `CKEDITOR` path to look in a local source it started to load files automatically from local, and meantime same css files was being loaded by rollup into bundled **main.js** and **main.mjs** files.

The idea behind this configuration is to rewrite every `.css` file during compilation process and replace the content with emptiness .

With this way we will avoid duplications in our code, also network will not give error about not founded files. because the other option was not to include that files in dist folder at all.

Reply:
Correct, you can check [recursive-copy](https://www.npmjs.com/package/recursive-copy) library examples for more examples